From 731aca526daae32a23e71448f2c03b76ea075075 Mon Sep 17 00:00:00 2001
From: Hariharan K <quic_harihk@quicinc.com>
Date: Wed, 16 Aug 2023 14:59:22 +0530
Subject: [PATCH] PCI: qcom: Set max payload size 256 bytes for IPQ9574/IPQ5332

This patch sets 256 bytes as payload size for IPQ9574/IPQ5332.
This allows PCIe RC to use the max payload size when a capable
link partner is connected.

Change-Id: I5fd39521c644e40de6bb938262f4b1d2c10b44ad
Signed-off-by: Hariharan K <quic_harihk@quicinc.com>
---
 drivers/pci/controller/dwc/pcie-qcom.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/pci/controller/dwc/pcie-qcom.c b/drivers/pci/controller/dwc/pcie-qcom.c
index 58f113adb512..cac472290310 100644
--- a/drivers/pci/controller/dwc/pcie-qcom.c
+++ b/drivers/pci/controller/dwc/pcie-qcom.c
@@ -1485,6 +1485,11 @@ static int qcom_pcie_post_init_1_27_0(struct qcom_pcie *pcie)
 	u32 val;
 	int i;
 
+	val = readl(pci->dbi_base + offset + PCI_EXP_DEVCTL);
+	val &= ~PCI_EXP_DEVCTL_PAYLOAD;
+	val |= PCI_EXP_DEVCTL_PAYLOAD_256B;
+	writel(val, pci->dbi_base + offset + PCI_EXP_DEVCTL);
+
 	writel(0x8000000, pcie->parf + PCIE20_v3_PARF_SLV_ADDR_SPACE_SIZE);
 
 	val = readl(pcie->parf + PCIE20_PARF_PHY_CTRL);
-- 
2.34.1

